Move the whole implementation into the header, not just a forward declaration,
otherwise include/rtl/ref.hxx will complain about an incomplete type.

--- libreoffice-7.2.4.1/include/svx/xmleohlp.hxx~	Fri Dec  3 00:10:29 2021
+++ libreoffice-7.2.4.1/include/svx/xmleohlp.hxx	Mon Jan  3 12:50:30 2022
@@ -33,6 +33,27 @@
 #include <memory>
 #include <optional>
 
+#include <com/sun/star/io/XStream.hpp>
+#include <unotools/streamwrap.hxx>
+#include <unotools/tempfile.hxx>
+#include <svtools/embedhlp.hxx>
+#include <unotools/ucbstreamhelper.hxx>
+#include <comphelper/storagehelper.hxx>
+#include <comphelper/embeddedobjectcontainer.hxx>
+#include <comphelper/fileformat.h>
+#include <cppuhelper/exc_hlp.hxx>
+#include <cppuhelper/implbase.hxx>
+
+using namespace ::osl;
+using namespace ::cppu;
+using namespace ::utl;
+using namespace ::com::sun::star;
+using namespace ::com::sun::star::document;
+using namespace ::com::sun::star::uno;
+using namespace ::com::sun::star::container;
+using namespace ::com::sun::star::io;
+using namespace ::com::sun::star::lang;
+
 enum class SvXMLEmbeddedObjectHelperMode
 {
     Read, Write
@@ -41,9 +62,29 @@
 namespace comphelper { class IEmbeddedHelper; }
 
 class SvGlobalName;
-class OutputStorageWrapper_Impl;
+//class OutputStorageWrapper_Impl;
 
+class OutputStorageWrapper_Impl : public ::cppu::WeakImplHelper<XOutputStream>
+{
+    ::osl::Mutex    maMutex;
+    Reference < XOutputStream > xOut;
+    TempFile aTempFile;
+    bool bStreamClosed : 1;
+    SvStream* pStream;
 
+public:
+    OutputStorageWrapper_Impl();
+
+// css::io::XOutputStream
+    virtual void SAL_CALL writeBytes(const Sequence< sal_Int8 >& aData) override;
+    virtual void SAL_CALL flush() override;
+    virtual void SAL_CALL closeOutput() override;
+
+    SvStream*   GetStream();
+};
+
+
+
 class SVXCORE_DLLPUBLIC SvXMLEmbeddedObjectHelper final :
     public cppu::WeakComponentImplHelper< css::document::XEmbeddedObjectResolver, css::container::XNameAccess >
 {
--- libreoffice-7.2.4.1/svx/source/xml/xmleohlp.cxx~	Fri Dec  3 00:10:29 2021
+++ libreoffice-7.2.4.1/svx/source/xml/xmleohlp.cxx	Mon Jan  3 12:36:47 2022
@@ -61,7 +61,7 @@
 #define XML_EMBEDDEDOBJECT_URL_BASE     "vnd.sun.star.EmbeddedObject:"
 #define XML_EMBEDDEDOBJECTGRAPHIC_URL_BASE "vnd.sun.star.GraphicObject:"
 
-
+/*
 class OutputStorageWrapper_Impl : public ::cppu::WeakImplHelper<XOutputStream>
 {
     ::osl::Mutex    maMutex;
@@ -80,6 +80,7 @@
 
     SvStream*   GetStream();
 };
+*/
 
 OutputStorageWrapper_Impl::OutputStorageWrapper_Impl()
     : bStreamClosed( false )
